streams/stream.go

```diff
@@ -54,6 +54,17 @@ func FromArray(array interface{}, threads ...int) *Stream {
 	return FromIterable(NewCollectionFromArray(array), threads...)
 }
 
+// FromMap Creates a Stream from a given map.
```

Comment:
Could you also add the map case in the `From` function to properly use the correct constructor?

```Go
func From(collection interface{}, threads ...int) *Stream {
	colReflectType := reflect.TypeOf((*IIterable)(nil)).Elem()

	if reflect.PtrTo(reflect.TypeOf(collection)).Implements(colReflectType) {
		return FromIterable(collection.(IIterable), threads...)
	}
	if reflect.TypeOf(collection).Kind() == reflect.Map {
		return FromMap(collection, threads...)
	}
	return FromArray(collection, threads...)
}```